pkg/controllers/provisioning/scheduling/nodeclaimtemplate.go

```diff
@@ -66,8 +66,11 @@ func (i *NodeClaimTemplate) ToNodeClaim(nodePool *v1beta1.NodePool) *v1beta1.Nod
 	nc := &v1beta1.NodeClaim{
 		ObjectMeta: metav1.ObjectMeta{
 			GenerateName: fmt.Sprintf("%s-", i.NodePoolName),
-			Annotations:  lo.Assign(i.Annotations, map[string]string{v1beta1.NodePoolHashAnnotationKey: nodePool.Hash()}),
-			Labels:       i.Labels,
+			Annotations: lo.Assign(i.Annotations, map[string]string{
+				v1beta1.NodePoolHashAnnotationKey:        nodePool.Annotations[v1beta1.NodePoolHashAnnotationKey],
```

Comment:
I think this is slightly problematic. What happens if we create a NodeClaim before we have updated the version and NodePool hash on controller startup? This NodeClaim will have the old hash and version. Further, it's possible that when we do the List operation, we don't get this NodeClaim (due to a race condition) and then we don't update the hash and version on this NodeClaim. That leaves some NodeClaims on the cluster with the old hash and version when we didn't mean to.

Reply:
This was from an older iteration, changed back to include the `nodePool.Hash()`